api/types/plugin.go

```diff
@@ -111,6 +113,25 @@ func (p *PluginV1) CheckAndSetDefaults() error {
 		if (bearer.Token == "") == (bearer.TokenFile == "") {
 			return trace.BadParameter("exactly one of Token and TokenFile must be specified")
 		}
+	case *PluginSpecV1_Okta:
+		// Check settings.
+		if settings.Okta == nil {
+			return trace.BadParameter("settings must be set")
+		}
+		if err := settings.Okta.CheckAndSetDefaults(); err != nil {
+			return trace.Wrap(err)
+		}
+
+		if p.Credentials == nil {
+			return trace.BadParameter("credentials must be set")
+		}
+		bearer := p.Credentials.GetBearerToken()
+		if bearer == nil {
+			return trace.BadParameter("okta plugin must be used with the bearer token credential type")
+		}
+		if (bearer.Token == "") == (bearer.TokenFile == "") {
```

Comment:
huh, that's a clever way to do it 👍

Reply:
I can't take credit for it. :-) Copy paste job!